### DISCUSSION TITLE
[2024-10-14] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#11313] [Weekly Demo Day]
- James: Widget support for Data.fetch and Data.post. #11199
- James: Fix signing yet again. #11206
- James: Widget for Database.connect #11216
- James: Small Fixes #11236
- James: Remove some catch alls #11254
- James: Some Additional Error checks for Enso_File #11275
- Radosław: Fix cloud tests by using correct config from the repo #11223
- Radosław: Resolving `~` in Enso asset paths to user home #11235
- Radosław: Run only relevant tests in the Cloud workflow #11249
- Radosław: Accessing and modifying description and labels of Enso Cloud assets #11255
- Radosław: Log failure to refresh the Cloud tokens #11261
- Radosław: Clean up credentials file after E2E job. #11285
- Radosław: When connecting to a Postgres database through a datalink stored on Enso Cloud, its asset ID is included in the audit logs #11291
- Radosław: Enable audit logs for Snowflake backend #11306 (and fix failing Snowflake tests at the same time)
- Greg: Extend Column_Numeric benchmarks to large ints and floats, and decimals #11134
- Greg: Extend the range of int/float arguments to outside the range of Java `long`, in `ceil`, `floor`, and `truncate` #11135
- Greg: Add `Table` and `Warnings` to standard exports of Visualization/Main.enso #11190
- Greg: Add format parameter to `Decimal.parse` #11205
- Greg: Add format parameter to Float.parse #11229
- Ad: Refactor unsupported db error messages #11156
- Ad: Add DB_Table feature flags #11163
- Ad: Update SQLServer version #11177
- Ad: Add graph editor and add to graph editor icons #11208
- Ad: Add Coalesce tests and fix In-Memory version #11242
- Ad: Update running tests documentation #11257
- Ad: Enable SQLServer select columns #11270
- Ad: Refactor Dialect_Flag #11273
- Ad: Remove .LCK file and prevent future ones #11286
- Martha: Scatterplot: Add filter down node functionality #11162
- Martha: use new icon for adding new nodes #11213
- Martha: Wip/mk/scatterplot custom axis #11225
- Martha: Full white space rendering and links fix #11238
- Martha: add method for xml doc drilldown #11258
- Martha: add error message #11284
- Martha: Add data quality indicators to table Viz #11307
- Hubert: Don't send expression updates about interrupts #11218
- Hubert: Do not run visualizations on InterruptedException #11250
- Hubert: Add retries to Data.read #11269
- Hubert: Cleanup list of Enso's project aggregates #11170
- Hubert: Don't coerce arg passed to electron #11200
- Hubert: Eliminate Akka's shutdown hook #11202
- Pavel: Speedup DataflowError.withDefaultTrace #11153
- Pavel: Fix MacOS Engine CI JVM tests by fixing directory-watcher module config #11188
- Pavel: Migrate some passes to Mini passes #11191
- Pavel: Fix booting enso on MacOS #11212
- Pavel: Show only fields from current atom constructor in the debugger #11217
- Pavel: [PoC] Generate IR definitions with annotation processor #11267
- Pavel: [5% smaller caches on average](https://github.com/enso-org/enso/pull/11303#issuecomment-2407366513) - some even **28%** smaller. #11303
- Dmitry: Run node in a different execution environment #11173
- Dmitry: Allow project manager to read files #11204
- Dmitry: Eliminate scala.Some allocations #11259

**Comment by JaroslavTulach** [comment]
- Test parallelism in EnsoParser. Using `synchronized` to avoid it. #11174
- Enso contribution to GraalVM merged into `24.2` version: https://github.com/oracle/graal/pull/8266
- Enso contribution to Snakeyaml allows us to remove hacks: #11207
- Helidon released new version, so we can compile `Ydoc-server` by _native image_
- Native image launcher size down to **150MB** (previously **420MB**) #11279
- Special handling of last signature in a block #11289
- Co-effort to use _mini passes_: #11191
- Avoid dependency on org.graalvm.collections #11107
- Don't convert type to the same type #11122
- Initialize Enso Terminal in a background thread #11149
- Avoid NPE on missing @anno name #11224
- Eliminate Blanks inside of Type.Ascription #11155
- Compute the exception message sooner than context is closed #11241
- Special treatment of `PanicSentinel` in `CatchPanicNode` #11167
- Use `IR.STRUCTURE_COMPARATOR` to compare two IR objects #11178
- Adjusting to new IGV 1.20 #11195
- Removing serialization metadata #11128
